frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -23,28 +23,59 @@ import {
     FileUpload,
     Toggle,
     FormField,
+    Multiselect,
 } from '@cloudscape-design/components';
 import React, { useEffect, useState } from 'react';
 import { useAppDispatch, useAppSelector } from '../../config/store';
 import { appConfig, appConfigList, getConfiguration, listConfigurations, updateConfiguration } from './configuration-reducer';
-import { IAppConfiguration } from '../../shared/model/app.configuration.model';
-import { useValidationReducer } from '../../shared/validation';
+import { Application, IAppConfiguration } from '../../shared/model/app.configuration.model';
+import { issuesToErrors, scrollToInvalid, useValidationReducer } from '../../shared/validation';
 import { z } from 'zod';
 import NotificationService from '../../shared/layout/notification/notification.service';
+import { emrApplications, listEMRApplications } from '../emr/emr.reducer';
+import { formatDisplayNumber } from '../../shared/util/form-utils';
+import { ClusterTypeConfiguration } from './cluster-sizes';
 
 export function DynamicConfiguration () {
     const applicationConfig: IAppConfiguration = useAppSelector(appConfig);
     const configList: IAppConfiguration[] = useAppSelector(appConfigList);
+    const applicationList: string[] = useAppSelector(emrApplications);
```

Comment:
There is too much name-overlap in the useEffect. I would name this `emrApplications` or `emrApplicationsLIst`